11_notify/willshen8/pkg/puppy/lostpuppy_test.go

```diff
@@ -0,0 +1,71 @@
+package puppy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func runTest(t *testing.T, payload []byte, expectedBody string, expectedHTTPCode int) {
+	req, err := http.NewRequest("POST", "/api/lostpuppy/", bytes.NewBuffer(payload))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// setup router
+	router := SetupRouter()
+
+	//setup routes
+	SetupLostPuppyRoutes(router)
+	// Setup recorder
+	rr := httptest.NewRecorder()
+
+	// Execute
+	router.ServeHTTP(rr, req)
+
+	// Get actual response
+	resp := rr.Result()
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
+	bodyString := string(body)
+
+	// check for http status
+	assert.Equal(t, expectedHTTPCode, resp.StatusCode)
+	// check for body
+	assert.Equal(t, expectedBody, bodyString)
+}
+
+func TestLostPuppyAPI(t *testing.T) {
+	tests := []struct {
+		testName    string
+		payload     []byte
+		HTTPCode    int
+		expectedMsg string
+	}{
+		{
+			testName:    "Test POST with nil payload",
+			payload:     nil,
+			HTTPCode:    http.StatusUnprocessableEntity,
+			expectedMsg: "Unprocessable Entity\n"},
```

Comment:
nit: put the `},` at the end on a line of its own. That way, every line inside the struct body is the same - they all end with a comma. If you later add another field to the struct, the you just add a single line to the body of these tests, rather than needing to modify the line with `},` as well. This makes the diff more readable when reviewing.

It also gives visual symmetry that makes it easier to scan the code.

Reply:
100% agree it's more readable this way.